crates/iroha_torii/src/lib.rs

```diff
@@ -239,79 +238,28 @@ impl Torii {
         ))
     }
 
-    /// Start main API endpoints.
-    ///
-    /// # Errors
-    /// Can fail due to listening to network or if http server fails
-    async fn start_api(
-        self: Arc<Self>,
-        shutdown_signal: &ShutdownSignal,
-    ) -> eyre::Result<Vec<task::JoinHandle<eyre::Result<()>>>> {
-        let torii_address = self.address.value();
-
-        let handles = torii_address
-            .to_socket_addrs()?
-            .map(TcpListener::bind)
-            .collect::<FuturesUnordered<_>>()
-            .try_collect::<Vec<TcpListener>>()
-            .await?
-            .into_iter()
-            .map(|listener| {
-                let signal = shutdown_signal.clone();
-                let signal = async move { signal.receive().await };
-                let torii = Arc::clone(&self);
-                let api_router = torii.create_api_router();
-
-                let serve_fut = async move {
-                    axum::serve(listener, api_router)
-                        .with_graceful_shutdown(signal)
-                        .await
-                        .map_err(eyre::Report::from)
-                };
-                task::spawn(serve_fut)
-            })
-            .collect();
-
-        Ok(handles)
-    }
-
     /// To handle incoming requests `Torii` should be started first.
     ///
     /// # Errors
     /// Can fail due to listening to network or if http server fails
-    #[iroha_futures::telemetry_future]
-    pub async fn start(
-        self,
-        shutdown_signal: &ShutdownSignal,
-    ) -> error_stack::Result<impl core::future::Future<Output = ()>, eyre::Report> {
-        let torii = Arc::new(self);
-        let mut handles = vec![];
-
-        handles.extend(
-            Arc::clone(&torii)
-                .start_api(shutdown_signal)
-                .await
-                .into_report()
-                .map_err(|err| err.attach_printable(torii.address.clone().into_attachment()))?,
-        );
-
-        let run = handles
-            .into_iter()
-            .collect::<FuturesUnordered<_>>()
-            .for_each(|handle| {
-                match handle {
-                    Err(error) => {
-                        iroha_logger::error!(%error, "Join handle error");
-                    }
-                    Ok(Err(error)) => {
-                        iroha_logger::error!(%error, "Error while running torii");
-                    }
-                    _ => {}
-                }
-                futures::future::ready(())
-            });
-
-        Ok(run)
+    // #[iroha_futures::telemetry_future]
+    pub async fn start(self, shutdown_signal: ShutdownSignal) -> error_stack::Result<(), Error> {
```

Comment:
Any plans to implement `telemetry_future`?

Reply:
I don't have any plans related to it =)

Created https://github.com/hyperledger/iroha/issues/5134